all_test.go

```diff
@@ -3501,3 +3502,67 @@ COMMIT;
 	}
 	wg.Wait()
 }
+
+func TestIssue136(t *testing.T) {
+	RegisterMemDriver()
+	db, err := sql.Open("ql-mem", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+	tx, err := db.Begin()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = tx.Exec(`
+create table category (key int, name string);
+create table condition (key int, name string);
+create table product (key int, catkey int, condkey int);
+
+insert into category values (1, "foo"), (2, "hello");
+insert into condition values (1, "bar"), (2, "baz");
+insert into product values (1, 1, 1), (2, 2, null), (3, null, 2), (4, null, null);
+	`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = tx.Commit()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rows, err := db.Query(`
+select *
+from
+  (select
+   product.key as product_key,
+   category.name as category_name,
+   product.condkey as product_condkey
+   from
+     product
+   left join category on category.key == product.catkey)
+left join condition on condition.key == product_condkey;
+	`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	c, _ := rows.Columns()
+	e := []string{
+		"product_key",
+		"category_name",
+		"product_condkey",
+		"condition.key",
+		"condition.name"}
+	if !reflect.DeepEqual(e, c) {
```

Comment:
Feel free to ignore, it's not critical: It's just few lines of non-reflect code that can replace reflect.DeepEqual in this simple case.